File: Legacy/ERP.Entities/Consignee.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Security.Policy;
using System.Text;

namespace ERP.Entities
{
    public class Consignee
    {
        public int ConId { get; set; }
        public string FConId { get; set; }
        public string ConFirstname { get; set; }
        public string ConLastname { get; set; }
        public int CivId { get; set; }
        public KeyValue Civility { get; set; }
        public string ConCode { get; set; }
        public string ConAdresseTitle { get; set; }
        public string ConAddress1 { get; set; }
        public string ConAddress2 { get; set; }
        public string ConAddress3 { get; set; }
        public string ConPostcode { get; set; }
        public string ConCity { get; set; }
        public string ConCountry { get; set; }
        public string ConTel1 { get; set; }
        public string ConTel2 { get; set; }
        public string ConFax { get; set; }
        public string ConCellphone { get; set; }
        public string ConEmail { get; set; }
        public bool ConRecieveNewsletter { get; set; }
        public string ConNewsletterEmail { get; set; }
        public bool ConIsDeliveryAdr { get; set; }
        public bool ConIsInvoicingAdr { get; set; }
        public int UsrCreatedBy { get; set; }
        public User UserCreator { get; set; }
        public DateTime DateCreation { get; set; }
        public DateTime DateUpdate { get; set; }
        public int SocId { get; set; }
        public string ConComment { get; set; }
        public int? ConCmuId { get; set; }
        public string ConCompanyname { get; set; }

    }
}

File: frontend/src/types/consignee.ts
/**
 * Consignee entity (delivery/invoicing address)
 */
export interface Consignee {
  con_id: number
  con_code?: string
  con_firstname?: string
  con_lastname?: string
  civ_id?: number
  con_adresse_title?: string
  con_address1?: string
  con_address2?: string
  con_address3?: string
  con_postcode?: string
  con_city?: string
  con_province?: string
  con_country?: string
  con_tel1?: string
  con_tel2?: string
  con_fax?: string
  con_cellphone?: string
  con_email?: string
  con_recieve_newsletter?: boolean
  con_newsletter_email?: string
  con_is_delivery_adr?: boolean
  con_is_invoicing_adr?: boolean
  usr_created_by?: number
  soc_id?: number
  con_d_creation?: string
  con_d_update?: string
  con_comment?: string
  con_cmu_id?: number
  con_company_name?: string
}

/**
 * DTO for creating a consignee
 */
export interface ConsigneeCreateDto {
  con_firstname?: string
  con_lastname?: string
  civ_id?: number
  con_code?: string
  con_adresse_title?: string
  con_address1?: string
  con_address2?: string
  con_address3?: string
  con_postcode?: string
  con_city?: string
  con_province?: string
  con_country?: string
  con_tel1?: string
  con_tel2?: string
  con_fax?: string
  con_cellphone?: string
  con_email?: string
  con_recieve_newsletter?: boolean
  con_newsletter_email?: string
  con_is_delivery_adr?: boolean
  con_is_invoicing_adr?: boolean
  usr_created_by?: number
  soc_id?: number
  con_comment?: string
  con_cmu_id?: number
  con_company_name?: string
}

/**
 * DTO for updating a consignee
 */
export interface ConsigneeUpdateDto extends ConsigneeCreateDto {}

/**
 * Consignee search parameters
 */
export interface ConsigneeSearchParams {
  search?: string
  soc_id?: number
  con_firstname?: string
  con_comment?: string
  con_email?: string
  con_postcode?: string
  con_city?: string
  con_address?: string
  con_company_name?: string
  con_tel?: string
  con_is_delivery_adr?: boolean
  con_is_invoicing_adr?: boolean
  page?: number
  pageSize?: number
  sort_by?: string
  sort_order?: 'asc' | 'desc'
}
